### DISCUSSION TITLE
Dynamic payloads in variable

### DISCUSSION THREAD

**Discussion by TimMaasGeesteranus** [projectdiscovery/nuclei#5755] [Ideas]
### Please describe your feature request:
In my opinion it would be very nice to have the ability to pass payloads into variables like this:

`-var requesturl="/users/{{id}}"`

where id is defined in the template like this:

```
path:
  - "{{BaseURL}}{{requesturl}}

payloads:
  id: payloads/id.txt
```

### Describe the use case of this feature:
This feature would allow for more flexibility in executing injection attacks on multiple or changing request urls while keeping the same payload input

**Comment by GeorginaReeder** [comment]
Hey @TimMaasGeesteranus , thanks for this!

I've gone ahead and created an issue/feature request which references this discussion, which can be found here: https://github.com/projectdiscovery/nuclei/issues/5758

Closing this topic in favour of the FR!
